### PR TITLE
Cleanup cpp codebase

### PR DESCRIPTION
**Pull request by hokein**
**No functionality change**: This PR cleans up electron cpp codebase (fixes a few code-style violations and uses C++11 features).

### CODE REVIEW

File: atom/common/platform_util_mac.mm [hunk 1]
```diff
@@ -71,11 +72,11 @@
 // thread safe, including LSGetApplicationForURL (> 10.2) and
 // NSWorkspace#openURLs.
 std::string OpenURL(NSURL* ns_url, bool activate) {
-  CFURLRef openingApp = NULL;
-  OSStatus status = LSGetApplicationForURL((CFURLRef)ns_url,
-                                           kLSRolesAll,
-                                           NULL,
-                                           &openingApp);
+  CFURLRef openingApp = nullptr;
+    OSStatus status = LSGetApplicationForURL(base::mac::NSToCFCast(ns_url),
```
Comment (deepak1556) [inline atom/common/platform_util_mac.mm]: nit: extra space.

Comment (hokein) [inline atom/common/platform_util_mac.mm]: Done.

File: atom/browser/native_window_mac.mm [hunk 2]
```diff
@@ -1358,22 +1358,22 @@ static bool FromV8(v8::Isolate* isolate, v8::Handle<v8::Value> val,
     // they are available in the minimum SDK version
     if (type == "selection") {
       // NSVisualEffectMaterialSelection
-      vibrancyType = (NSVisualEffectMaterial) 4;
+      vibrancyType = static_cast<NSVisualEffectMaterial>( 4);
```
Comment (thomasjo) [inline atom/browser/native_window_mac.mm]: The leading space in front of the number here (and below) seems like an oversight.

Comment (hokein) [inline atom/browser/native_window_mac.mm]: oops, didn't note that. Done.

**Comment by kevinsawicki** [comment]
Thanks @hokein 👍

**Pull request merged**